src/test/scala/org/broadinstitute/dsde/firecloud/service/StatusServiceSpec.scala

```diff
@@ -0,0 +1,149 @@
+import org.broadinstitute.dsde.firecloud.{Application, FireCloudConfig}
+import org.broadinstitute.dsde.firecloud.mock.MockUtils
+import org.broadinstitute.dsde.firecloud.model.{SubsystemStatus, SystemStatus}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol.impSystemStatus
+import org.broadinstitute.dsde.firecloud.webservice.StatusApiService
+import org.mockserver.integration.ClientAndServer
+import org.mockserver.integration.ClientAndServer.startClientAndServer
+import org.mockserver.model.HttpError
+import org.mockserver.model.HttpRequest.request
+import spray.http.StatusCodes.{InternalServerError, OK}
+import spray.routing.HttpService
+import spray.httpx.SprayJsonSupport._
+
+
+/**
+  * Created by anichols on 4/13/17.
+  */
+class StatusServiceSpecMockServer extends BaseServiceSpec with HttpService with StatusApiService {
+
+  def actorRefFactory = system
+
+  // Typically, the BaseServiceSpec provides an `app: Application` member that has all Mock DAOs.
+  // Here, we want to send requests through the Agora DAO to the Mock Server, so create a new Application
+  // with the HTTP DAO instead of Mock.
+  val customApp = new Application(new HttpAgoraDAO(FireCloudConfig.Agora), googleServicesDao, ontologyDao, rawlsDao, searchDao, thurloeDao)
+  val statusServiceConstructor: () => StatusService = StatusService.constructor(customApp)
+
+  var agoraServer: ClientAndServer   = startClientAndServer(MockUtils.methodsServerPort) // Agora = methods
+
+  override def afterAll(): Unit = {
+    agoraServer.stop()
+  }
+
+  agoraServer.when(
+    request().withMethod("GET").withPath("/status")
+  ).respond(
+    org.mockserver.model.HttpResponse.response()
+      .withHeaders(MockUtils.header)
+      .withBody("""{ "status": "down", "message": ["Agora is down"] }""")
+  )
+
+  "StatusService indicates Agora failure when \"down\" response" in {
+    Get("/status") ~> sealRoute(publicStatusRoutes) ~> check {
+      status.intValue should be(InternalServerError.intValue)
+      val response = responseAs[SystemStatus]
+      response.ok should be(false)
+      response.systems("Agora").ok should be(false)
+      response.systems("Agora").messages should be(Some(List("Agora is down")))
+      response.systems.size should be(4)
+    }
+  }
+
+  agoraServer.when(
+    request().withMethod("GET").withPath("/status")
+  ).respond(
+    org.mockserver.model.HttpResponse.response()
+      .withHeaders(MockUtils.header)
+      .withBody("""{ "status": "up", "message": ["No problems with Agora"] }""")
+  )
+
+  "StatusService indicates OK" in {
+    Get("/status") ~> sealRoute(publicStatusRoutes) ~> check {
+      status.intValue should be(OK.intValue)
+      val response = responseAs[SystemStatus]
+      response.ok should be(true)
+      response.systems("Agora").ok should be(true)
+      response.systems("Agora").messages should be(None)
+      response.systems.size should be(4)
+    }
+  }
```

Comment:
you're modifying what the mock server returns for each test in this class. I'd prefer to see the mockserver expectation set up inside of each test's closure, like this:
```
  "StatusService indicates OK" in {
    agoraServer.when(
      request().withMethod("GET").withPath("/status")
    ).respond(
      org.mockserver.model.HttpResponse.response()
        .withHeaders(MockUtils.header)
        .withBody("""{ "status": "up", "message": ["No problems with Agora"] }""")
    )
    Get("/status") ~> sealRoute(publicStatusRoutes) ~> check {
      status.intValue should be(OK.intValue)
      val r = responseAs[SystemStatus]
      r.ok should be(true)
      r.systems("Agora").ok should be(true)
      r.systems("Agora").messages should be(None)
      r.systems.size should be(4)
    }
```

BUT, this causes failures because tests are also not cleaning up their expectations. Try mixing in the `BeforeAndAfterEach` trait, and add:
```
  override def beforeEach(): Unit = {
    agoraServer.reset()
  }
```
to ensure the server is clean.